src/app.js

```diff
@@ -1,21 +1,38 @@
 'use strict';
 
-/**
- * Implement sum function:
- *
- * Function takes 2 numbers and returns their sum
- *
- * sum(1, 2) === 3
- * sum(1, 11) === 12
- *
- * @param {number} a
- * @param {number} b
- *
- * @return {number}
- */
-function sum(a, b) {
-  // write code here
-  return a + b;
-}
-
-module.exports = sum;
+const { createServer } = require('http');
+const { URL: NodeURL } = require('url');
+const fs = require('fs/promises');
+const path = require('path');
+
+createServer(async(req, res) => {
+  res.setHeader('Content-type', 'application/json');
+
+  const newUrl = new NodeURL(req.url, `http://${req.headers.host}`);
+  const pathWithoutFile = newUrl.pathname.replace('/file/', '');
+  const publicFolderPath = path.resolve(__dirname, '..', 'public');
+
+  if (pathWithoutFile.length && newUrl.pathname.slice(0, 6) === '/file/') {
```

Comment:
The server might allow accessing files outside the 'public' directory, due to not checking if the resulting full path is actually within the 'public' directory after resolving '..' and '.' paths. You should add a check to ensure that the resulting full path is within the 'public' directory.